### PR TITLE
OMP corrections

### PR DESCRIPTION
**Pull request by mand35**

### CODE REVIEW

File: _lessons/python-scatter/04b-openmp.md [hunk 1]
```diff
@@ -19,9 +19,9 @@ cd openmp
 
 ## Why implement OpenMP parallelisation
 
-One way to speed up your application is using the available resources more efficiently. This approach was used while porting our Python code to C++ by removing the interpreter's overhead. Here we will improve performance by using more resources instead.
+One way to speed up your application is using the available resources more efficiently. This approach was used while porting our Python code to C/C++ by removing the interpreter's overhead. Here we will improve performance by using more resources instead.
 
-Most modern computers have multi-core CPUs and we can use two or more of these cores to execute instructions in parallel. All cores can access the same, shared memory.
+Most modern computers have multi-core CPUs and we can use two or more of these cores to execute instructions in parallel. All cores in one CPU can access the same, shared memory.
```
Comment (pletzer) [inline _lessons/python-scatter/04b-openmp.md]: All cores of a CPU can access the same, shared memory. Furthermore, all the CPUs belonging to a node also share the same memory. Given that supercomputers often have tens of cores per node, anywhere between 20-80 instructions can in principle be executed in parallel.

File: _lessons/python-scatter/04b-openmp.md [hunk 2]
```diff
@@ -102,7 +102,7 @@ double mySumSq(int n, double* arr) {
 }
 ```
 
-With the `parallel` statement we ask the compiler to spawn threads. The number of threads can be set using environment variable `OMP_NUM_THREADS`, which can be anything between 1 and the number of cores on a node, e.g., `export OMP_NUM_THREADS=36`.
+With the `parallel` statement we ask the compiler to spawn threads. The number of threads can be set *during runtime* using environment variable `OMP_NUM_THREADS`, which can be anything between 1 and the number of cores on a node, e.g., `export OMP_NUM_THREADS=36`. Note, typically application do not scale that far.
```
Comment (pletzer) [inline _lessons/python-scatter/04b-openmp.md]: Note, many applications may not scale to the maximum number of cores on a node due to load balancing and Amdahl's law, which states that the maximum speedup is limited by the ratio of parallel to serial parts of the code. As a rule of thumb, if 5 percent of the time is spent in part of the code that cannot be parallelised (that is a critical section or serial part) then the maximum parallel speedup is about 20 (=1/0.05) and there is no point in using more than 20 threads.

**Review by pletzer — CHANGES_REQUESTED** [review]
Hi @mand35 
Thanks for adding "n" to the shared list. For C++ code it is not required as n is defined outside the scope but may be a good habit. 

Do we want a paragraph about scalability? I added a blurp.

File: _lessons/python-scatter/04b-openmp.md (hunk 2 above)
Comment (mand35) [reply _lessons/python-scatter/04b-openmp.md]: Thanks Alex for further suggestions I added the statements.

**Review by pletzer — APPROVED** [review]

**Pull request merged**